tests/rptest/tests/compatibility/kafka_streams_test.py

```diff
@@ -446,21 +424,187 @@ def test_kafka_streams_map_function(self):
 
         # Consume the data
         consumer.start()
+        consume(self._ctx, consumer, self.is_valid_msg)
+
+        consumer.stop()
+        producer.stop()
+        example.stop()
+
 
-        def try_cons():
-            i = 0
-            msgs = consumer.messages
-            while i < len(msgs) and not self.is_valid_msg(msgs[i]):
-                i += 1
+class KafkaStreamsInteractiveQueries(RedpandaTest):
+    """
+    Test KafkaStreams InteractiveQueries example which locates and
+    queries state stores.
+    """
 
-            return i < len(msgs)
+    topics = (TopicSpec(name="TextLinesTopic"), )
 
-        wait_until(
-            try_cons,
-            timeout_sec=TIMEOUT,
-            backoff_sec=5,
-            err_msg=f"kafka-streams {self._ctx.cls_name} consumer failed")
+    def __init__(self, test_context):
+        super(KafkaStreamsInteractiveQueries,
+              self).__init__(test_context=test_context)
+        self._ctx = test_context
 
-        consumer.stop()
+    def _http_get(self, host, route, cond):
+        def try_get():
+            res = requests.get(f"http://{host}:7070{route}")
+
+            self.logger.debug(f"Status code: {res.status_code}")
+            self.logger.debug(f"Text: {res.text}")
+
+            if not res.ok:
+                return False
+
+            return cond(res.text)
+
+        wait_until(try_get,
+                   timeout_sec=TIMEOUT,
+                   backoff_sec=5,
+                   err_msg=f"GET {route} failed")
+
+    @cluster(num_nodes=6)
+    def test_kafka_streams_interactive_queries(self):
+        # Create the two examples needed
+        # 7070 and 7071 are port numbers
+        example_jar1 = KafkaStreamExamples.InteractiveQueriesExample(
+            self.redpanda, 7070)
+        example_jar2 = KafkaStreamExamples.InteractiveQueriesExample(
+            self.redpanda, 7071)
+        example1 = ExampleRunner(self._ctx, example_jar1, timeout_sec=TIMEOUT)
+        example2 = ExampleRunner(self._ctx, example_jar2, timeout_sec=TIMEOUT)
```

Comment:
I create two instances of InteractiveQueriesExample because that's what the example docs do. I'm simply re-creating their use case in ducktape.

https://github.com/confluentinc/kafka-streams-examples/blob/7.0.0-post/src/main/java/io/confluent/examples/streams/interactivequeries/WordCountInteractiveQueriesExample.java